File: xamarin/common/Presenter/FavouritesPresenter.cs
using System;
using System.Collections.Generic;
using PropertyCross.Model;

namespace PropertyCross.Presenter
{
  /// <summary>
  /// A presenter which allows the user to view their favourites.
  /// </summary>
  public class FavouritesPresenter
  {
    /// <summary>
    /// The interface this presenter requires from the assocaited view.
    /// </summary>
    public interface View
    {
      void SetFavourites(List<Property> properties);

      event EventHandler<PropertyEventArgs> PropertySelected;
    }

    private PropertyCrossPersistentState _state;

    private INavigationService _navigationService;

    public FavouritesPresenter(INavigationService navigationService, PropertyCrossPersistentState state)
    {
      _state = state;
      _navigationService = navigationService;
    }

    public void SetView(View view)
    {
      view.SetFavourites(_state.Favourites);
      view.PropertySelected -= View_PropertySelected;
      view.PropertySelected += View_PropertySelected;
    }

    private void View_PropertySelected(object sender, PropertyEventArgs e)
    {
      var presenter = new PropertyPresenter(_state, e.Property);
      _navigationService.PushPresenter(presenter);
    }
  }
}

File: native/android/PropertyCross/app/src/main/java/com/propertycross/android/presenter/FavouritesPresenter.java
package com.propertycross.android.presenter;

import java.util.List;

import com.propertycross.android.events.Callback;
import com.propertycross.android.events.PropertyEventArgs;
import com.propertycross.android.events.PropertySelectedEvent;
import com.propertycross.android.model.Property;

public class FavouritesPresenter {

	public interface View {
		void setFavourites(List<Property> properties);
		void propertySelected(Callback<PropertySelectedEvent> callback);
	}

	private PropertyFinderPersistentState state;
	private INavigationService navigationService;

	public FavouritesPresenter(INavigationService navigationService, PropertyFinderPersistentState state) {
		this.state = state;
		this.navigationService = navigationService;
	}

	public void setView(View view) {
		view.setFavourites(state.getFavourites());
		view.propertySelected(new Callback<PropertySelectedEvent>() {
			
			public void complete(PropertySelectedEvent event) {
				PropertyPresenter presenter = 
						new PropertyPresenter(state,((PropertyEventArgs) event.getArguments()).getProperty());
				navigationService.pushPresenter(presenter);
			};
		});
	}
}
